hazelcast/src/main/java/com/hazelcast/nio/serialization/GenericRecordBuilder.java

```diff
@@ -68,775 +80,1025 @@ static GenericRecordBuilder compact(String typeName) {
+     *                                         trying to be set without using
+     *                                         {@link
+     *                                         GenericRecord#newBuilderWithClone()}.
      */
     @Nonnull
     GenericRecordBuilder setString(@Nonnull String fieldName, @Nullable String value);
 
     /**
-     * It is legal to set the field again only when Builder is created with {@link GenericRecord#cloneWithBuilder()}.
-     * Otherwise, it is illegal to set to the same field twice.
-     * This method allows nested structures. Subclass should also created as `GenericRecord`.
+     * It is legal to set the field again only when Builder is created with
+     * {@link GenericRecord#newBuilderWithClone()}. Otherwise, it is illegal to
+     * set to the same field twice. This method allows nested structures.
+     * Subclass should also be created as `GenericRecord`.
      *
-     * @param fieldName name of the field as it is defined in its class definition.
-     *                  See {@link ClassDefinition} for {@link Portable}
+     * @param fieldName name of the field as it is defined in its schema/class
+     *                  definition. It should be composed of only alphanumeric
+     *                  characters for {@link Portable} GenericRecords.
      * @param value     to set to GenericRecord
      * @return itself for chaining
-     * @throws HazelcastSerializationException if the field name does not exist in the class definition or
-     *                                         the type of the field does not match the one in the class definition or
-     *                                         Same field is trying to be set without using
-     *                                         {@link GenericRecord#cloneWithBuilder()}.
+     * @throws HazelcastSerializationException if the field name does not exist
+     *                                         in the schema/class definition or
+     *                                         the type of the field does not
+     *                                         match the one in the schema/class
+     *                                         definition or the same field is
+     *                                         trying to be set without using
+     *                                         {@link
+     *                                         GenericRecord#newBuilderWithClone()}.
      */
     @Nonnull
     GenericRecordBuilder setGenericRecord(@Nonnull String fieldName, @Nullable GenericRecord value);
 
     /**
-     * Sets a decimal which is arbitrary precision and scale floating-point number
-     * It is legal to set the field again only when Builder is created with {@link GenericRecord#cloneWithBuilder()}.
-     * Otherwise, it is illegal to set to the same field twice.
+     * Sets a decimal which is arbitrary precision and scale floating-point
+     * number It is legal to set the field again only when Builder is created
```

Comment:
```suggestion
     * Sets a decimal which is arbitrary precision and scale floating-point
     * number. It is legal to set the field again only when Builder is created
```

We don't have the first sentence in other methods: `Sets a decimal which is arbitrary precision and scale floating-point number.`. I think we should remove these from all of them or add to all of them. for you info, the first sentence only appears in decimal, datetime methods and arrays of them.

Reply:
I removed the ones from the arrays, as they don't offer any useful information. Regarding the ones in the decimal and date related ones, I think it is OK to have them as an extra description for those, as the method names and the parameters they have are not directly relatable. I don't see it as a problem to have an extra definition only on some of them